ports/raspberrypi/bindings/rp2clock/InputPin.c

```diff
@@ -0,0 +1,197 @@
+    enum { ARG_pin, ARG_index, ARG_src_freq, ARG_target_freq };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_pin,         MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_index,       MP_ARG_OBJ | MP_ARG_KW_ONLY,  {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_src_freq,    MP_ARG_INT | MP_ARG_KW_ONLY,  {.u_int = 0} },
+        { MP_QSTR_target_freq, MP_ARG_INT | MP_ARG_KW_ONLY,  {.u_int = 0} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    rp2clock_inputpin_obj_t *self = m_new_obj(rp2clock_inputpin_obj_t);
+    self->base.type = &rp2clock_inputpin_type;
+
+    // Validate pin number
+    common_hal_rp2clock_inputpin_validate_index_pin(args[ARG_pin].u_rom_obj);
+    self->pin = args[ARG_pin].u_rom_obj;
+
+    // Validate pin based on clock
+    if (args[ARG_index].u_rom_obj != mp_const_none) {
+        self->index = validate_index(args[ARG_index].u_rom_obj, MP_QSTR_index);
+        self->src_freq = args[ARG_src_freq].u_int;
+        self->target_freq = args[ARG_target_freq].u_int;
+        // Validate frequencies if set
+        if (self->src_freq && self->target_freq) {
+            common_hal_rp2clock_inputpin_validate_freqs(args[ARG_src_freq].u_int, args[ARG_target_freq].u_int);
+        }
+    } else {
+        self->index = INDEX_NONE;
+    }
+    self->enabled = false;
+    return MP_OBJ_FROM_PTR(self);
+};
+
+//|     def deinit(self) -> None:
+//|         """Releases the pin and frees any resources."""
+STATIC mp_obj_t rp2clock_inputpin_deinit(mp_obj_t self_in) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    // Release pin
+    common_hal_rp2clock_inputpin_deinit(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(rp2clock_inputpin_deinit_obj, rp2clock_inputpin_deinit);
+
+//|     def enable(self) -> None:
+//|         """Configures the pin and enables the internal clock"""
+STATIC mp_obj_t rp2clock_inputpin_enable(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    check_for_deinit(self);
+    common_hal_rp2clock_inputpin_enable(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(rp2clock_inputpin_enable_obj, 1, rp2clock_inputpin_enable);
+
+//|     def disable(self) -> None:
+//|         """Disables the pin and internal clock"""
+STATIC mp_obj_t rp2clock_inputpin_disable(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    check_for_deinit(self);
+    common_hal_rp2clock_inputpin_disable(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(rp2clock_inputpin_disable_obj, 1, rp2clock_inputpin_disable);
+
+//|     def set_freq(self, src_freq: int, target_freq: int) -> None:
+//|         """Configures the src and target frequency. Must be set before enable() is called."""
```

Comment:
Is this just so you can change an existing InputPin? I think you can just delete this, enable and disable and just rely on deinit and reconstruct.

Reply:
I think you can remove `enable()`, `disable()` and `enabled` too. Instead, folks can just use the constructor and `deinit()`.

Reply:
Fixed on latest commit.

Reply:
There are use cases where you'd want to create an external pin object and then enable/disable the clock at runtime. For instance when putting the external chip in low power mode you shouldn't drive the clock to it. My use case actually requires this. This is compounded by the current issue where resources aren't released until garbage collection is called. My opinion is the functionality is required.

Reply:
`.deinit()` is the way to turn it off immediately instead of waiting for the finalizer. Do you want to keep the pin "in use" even when it isn't active?

Reply:
Good point on .deinit(). Since it's a dedicated hardware pin, I  think logically it should be use whether the clock is enabled or not. It's akin to driving a normal GPIO, you wouldn't want to release the object and create a new one just to change the value from low to high. The only difference is a high level puts out a clock.

Reply:
Ok, then you can leave the `.enabled` property but make it writable instead of having `enable()` and `disable()` functions. I'll mark the spot with a pointer to an example.

Reply:
Sounds good